ngraph/core/reference/src/runtime/reference/matrix_nms.cpp

```diff
@@ -0,0 +1,443 @@
+                                  });
+                    }
+                }
+
+                std::copy(num_per_batch.begin(), num_per_batch.end(), valid_outputs);
+                for (size_t i = 0; i < filtered_boxes.size(); i++)
+                {
+                    selected_indices[i] = filtered_boxes[i].index;
+                    auto selected_base = selected_outputs + i * 6;
+                    selected_base[0] = filtered_boxes[i].class_index;
+                    selected_base[1] = filtered_boxes[i].score;
+                    selected_base[2] = filtered_boxes[i].box.x1;
+                    selected_base[3] = filtered_boxes[i].box.y1;
+                    selected_base[4] = filtered_boxes[i].box.x2;
+                    selected_base[5] = filtered_boxes[i].box.y2;
+                }
+            }
+
+            void matrix_nms_postprocessing(void* prois,
+                                           void* pscores,
+                                           void* pselected_num,
+                                           const ngraph::element::Type output_type,
+                                           const std::vector<float>& selected_outputs,
+                                           const std::vector<int64_t>& selected_indices,
+                                           const std::vector<int64_t>& valid_outputs)
+            {
+                int64_t total_num = std::accumulate(valid_outputs.begin(), valid_outputs.end(), 0);
+
+                float* ptr = static_cast<float*>(prois);
+                memcpy(ptr, selected_outputs.data(), total_num * sizeof(float) * 6);
+
+                if (pscores)
+                {
+                    if (output_type == ngraph::element::i64)
+                    {
+                        int64_t* indices_ptr = static_cast<int64_t*>(pscores);
+                        memcpy(indices_ptr, selected_indices.data(), total_num * sizeof(int64_t));
+                    }
+                    else
+                    {
+                        int32_t* indices_ptr = static_cast<int32_t*>(pscores);
+                        for (size_t i = 0; i < (size_t)total_num; ++i)
+                        {
+                            indices_ptr[i] = static_cast<int32_t>(selected_indices[i]);
+                        }
+                    }
+                }
+
+                if (pselected_num)
+                {
+                    if (output_type == ngraph::element::i64)
+                    {
+                        int64_t* valid_outputs_ptr = static_cast<int64_t*>(pselected_num);
+                        std::copy(valid_outputs.begin(), valid_outputs.end(), valid_outputs_ptr);
+                    }
+                    else
+                    {
+                        int32_t* valid_outputs_ptr = static_cast<int32_t*>(pselected_num);
+                        for (size_t i = 0; i < (size_t)valid_outputs.size(); ++i)
```

Comment:
Maybe  we need to delete `(size_t)`, because `std::vector<T>::size()` returns a value with type `size_t`?

Reply:
Done